### PR TITLE
Add pagination functionality to get pipelines from projects

### LINKED ISSUE #61: pagination on client
# New feature request

implement pagination on client

Please enter a concise and clear description on what would you like to implement.

### Checklist for proposal reviewer

- [ ] The proposal is clear on what changes need to be made in the code base.

- [ ] The proposal is clear on the architectural changes that will be affected.

- [ ] The proposal is clear on the breaking changes and how to mitigate them.

- [ ] The proposal is clear on what the deployment/configuration strategy is.

- [ ] The proposal is clean on how testing will validate the feature implementation.

### Reviewers

- [ ] Reviewed by a team member of the @calyptia/backend team
- [ ] Reviewed by a product owner.

### Disclaimers

- If there are multiple solutions, please present each one separately.
- Save comparisons for the very end.
- Please do not create a Pull Request without creating an issue first.
- Any change needs to be discussed before proceeding.
- Failure to do so may result in the rejection of the pull request.

### PR DESCRIPTION
**Pull request by gabrielbussolo**
# Summary of this proposal

## Issue(s) number(s)

fix #61

## Checklist for submitter

- [ ] My PR has a related issue/bug number.
- [ ] My PR provides tests
  - [ ] Integration tests are added/passes
  - [ ] Unit tests are added/passes
  - [ ] End-to-end tests added/passes
  - [ ] Static code check added/passes
  - [ ] Linting on documentation added/passes
  - [ ] Does not affect code coverage stats
- [ ] My PR requires updating dependencies
- [ ] My PR has the documentation changes required.

## Checklist for reviewer

- [ ] The proposal fixes a bug/issue or implements a new feature that is well described.
- [ ] The proposal has sufficient test cases that covers the changes.
  - [ ] If changes an API, it does not break backwards compatibility (-1 major version).
- [ ] If integration is required, the proposal has integration tests.
- [ ] The proposal does not break coverage stats
- [ ] The proposal has the required documentation changes.

## Backport

- [ ] Backport to the latest stable release.

### CODE REVIEW

File: client/pipeline_config_test.go [hunk 1]
```diff
@@ -14,21 +16,42 @@ func TestClient_PipelineConfigHistory(t *testing.T) {
...
+		wantNoEqual(t, got.Items[1].ID, "")
+		wantEqual(t, got.Items[1].RawConfig, testFbitConfigWithAddr)
+		wantNoTimeZero(t, got.Items[1].CreatedAt)
+	})
+
+	t.Run("pagination", func(t *testing.T) {
+		for i := 0; i < 9; i++ {
+			_, err := asUser.UpdatePipeline(ctx, pipeline.ID, types.UpdatePipeline{
+				RawConfig: ptrStr(strings.Replace(testFbitConfigWithAddr, "24224", strconv.Itoa(24224+i), -1)),
+			})
```
Comment (niedbalski) [inline client/pipeline_config_test.go:44]: ^^ fix this please

File: client/membership_test.go [hunk 2]
```diff
@@ -20,4 +20,7 @@ func TestClient_Members(t *testing.T) {
 	wantEqual(t, got.Items[0].Roles, []types.MembershipRole{types.MembershipRoleCreator})
 	wantNoTimeZero(t, got.Items[0].CreatedAt)
 	wantNoEqual(t, got.Items[0].User, nil)
+	t.Run("pagination", func(t *testing.T) {
+		// TODO: HOW TO CREATE NEW USERS TO TEST THE PAGINATION?
```
Comment (nicolasparada) [inline client/membership_test.go:24]: Oh, that's a good question since you must complete the whole email joining flow 🤔

Comment (nicolasparada) [reply client/membership_test.go:24]: I suggest to leave this one out of the PR and leave the TODO comment to add the tests later. For now just test manually that it works.

Comment (niedbalski) [reply client/membership_test.go:24]: I agree with this.

**Pull request merged**